### DISCUSSION TITLE
Feature to pause and resume encoding

### DISCUSSION THREAD

**Discussion by tijder** [kokorin/Jaffree#392] [Ideas]
Hello thanks for creating this awesome library!

I had an idea for a nice feature. The feature to pause and resume a ffmpeg process. This can be archived by pausing the ffmpeg process.

I did a quick test on Linux how it could be done, https://github.com/ister-app/Jaffree/commit/ee77e507785d5e0544756f518ae0edefb4eb329c. To get the pid in Linux I needed to use Java 9+ functionality. It should also be possible to do on [Windows](https://github.com/Dusk-Labs/nightfall/blob/master/src/utils.rs#L37).

Do you like this feature? Are you open for a pull request if I create one?

**Comment by kokorin** [comment]
Hello, before starting to code, could you describe potential use cases? Is it to save resources for other encodings? Another question is how to pause ffmpeg reading live stream?

**Comment by tijder** [comment]
Thanks for your reply.

I'm building a small media server. Witch can serve my local mkv media files. For streaming to the browser I create with Jaffree a HLS media stream. The issue is that transcoding takes up a lot of resources of the server. When I change subtitle/audio stream, its need to start transcoding again.

I like the option that I can start the HLS transcode session and when the more then * minutes more is transcodes then the client is currently at. The transcode can be paused and when the client is close to the end of the stream it can resume the transcode session again.

So yes its to keep resources available for other (transcode) processes. 

I don't know if pausing a live steam is possible.
